src/main/java/com/sarality/db/InMemoryTable.java

```diff
@@ -41,8 +40,27 @@ public Long create(T data) {
 
   @Override
   public List<T> readAll(Query query) {
+
     List<T> dataList = new ArrayList<>();
-    dataList.addAll(dataMap.values());
+
+    //null query, just return everything
+    if (query == null) {
+      dataList.addAll(dataMap.values());
+      return dataList;
+    }
+
+    String[] whereArgs = query.getWhereClauseArguments();
+
+    //assumes that there is only 1 argument in the where clause which is a primary key Id to retrieve.
+    if (whereArgs.length != 0) {
+      Long key = Long.parseLong(whereArgs[0]);
+      dataList.add(dataMap.get(key));
+    }
+    else {
+      //return everything: no other type of query is supported
+      dataList.addAll(dataMap.values());
```

Comment:
Throw an exception. Take a look at comments in the CL that you closed. We should just make sure that we know about the PrimaryKey Column and just check for a where clause of type "PRIMARY_KEY_COLUMN = ?". If nothing else we should have a TODO for that